src/keylessBackup/encryption.ts

```diff
@@ -0,0 +1,91 @@
+import crypto from 'crypto'
+import { ec as EC } from 'elliptic'
+import hkdf from 'futoin-hkdf'
+
+const ec = new EC('secp256k1')
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function deriveKeyFromKeyShares(keyshare1: string, keyshare2: string): Buffer {
+  // Combining the keyshares
+  const combinedKeyShares = Buffer.concat([
+    Buffer.from(keyshare1, 'utf8'),
+    Buffer.from(keyshare2, 'utf8'),
+  ])
+
+  // Using futoin-hkdf to derive a 256-bit key
+  return hkdf(combinedKeyShares, 32, {
+    salt: 'some fixed salt',
+    info: 'encryption',
+    hash: 'SHA-256',
+  })
+}
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares then derives a secp256k1 pair from that.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function getSecp256K1KeyPair(keyshare1: string, keyshare2: string): EC.KeyPair {
+  const derivedKey = deriveKeyFromKeyShares(keyshare1, keyshare2)
+
+  return ec.keyFromPrivate(derivedKey)
+}
+
+/**
+ * Encrypts a passphrase using AES-256-GCM encryption with a key derived from two key shares.
+ *
+ * @param {string} keyshare1 - The first keyshare used to derive the encryption key.
+ * @param {string} keyshare2 - The second keyshare used to derive the encryption key.
+ * @param {string} passphrase - The passphrase to encrypt.
+ * @returns {string} The encrypted passphrase, formatted as `iv:encrypted:authTag`, all parts base64 encoded.
+ */
+export function encryptPassphrase(
+  keyshare1: string,
+  keyshare2: string,
```

Comment:
How are these keyshares encoded? Would it be possible to use a `Buffer` type instead? The reason I mention it is that it would be good to add some basic sanity check to this function such that it throws if they keys passed in are obviously bad. This would be helpful if something goes wrong in implementation and an `undefined` or empty key share gets passed in. I would recommend checking that the inputs are defined and have length of at least 16 bytes (so if this is a hex-encoded string, 32 characters).